include/Foundation/NSDecimal.h

```diff
@@ -1,62 +1,65 @@
-/* Copyright (c) 2007 Christopher J. W. Lloyd
-
-Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the
-"Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish,
-distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the
-following conditions:
-
-The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
-SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. */
-
-#ifndef _NSDECIMAL_H_
-#define _NSDECIMAL_H_
+//******************************************************************************
+//
+//  Copyright (c) 2007 Christopher J. W. Lloyd
+//  Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+#pragma once
 
 #include <limits.h>
 #import <Foundation/NSObjCRuntime.h>
 
-#define NSDecimalMaxSize (8)
-#define NSDecimalNoScale SHRT_MAX
-
 enum { NSRoundPlain, NSRoundDown, NSRoundUp, NSRoundBankers };
-
-typedef uint32_t NSRoundingMode;
+typedef NSUInteger NSRoundingMode;
```

Comment:
Since the official documentation has been gimped, we can no longer tell whether it was `NS_ENUM` or not.

Still, we have an opportunity to be more concise and better-specified than the reference platform. Wherever that's possible, I think we should.

---
In reply to: [80775800](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r80775800) [](ancestors = 80775800,80771005)